src/bundle_list.c

```diff
@@ -419,6 +445,91 @@ static enum swupd_code show_bundle_reqd_by(const char *bundle_name, bool server,
 	return ret;
 }
 
+static enum swupd_code list_orphaned_bundles(int version)
+{
+	enum swupd_code ret_code = SWUPD_OK;
+	struct list *tracked_bundles = NULL;
+	struct list *bundles = NULL;
+	struct list *subs = NULL;
+	struct list *iter = NULL;
+	struct manifest *mom = NULL;
+	struct file *bundle_manifest = NULL;
+	char *name = NULL;
+	int count = 0;
+	int ret;
+
+	progress_next_step("load_manifests", PROGRESS_UNDEFINED);
+	info("Loading required manifests...\n");
+	mom = load_mom(version, NULL);
+	if (!mom) {
+		error("Cannot load official manifest MoM for version %i\n\n", version);
+		return SWUPD_COULDNT_LOAD_MOM;
+	}
+
+	// start with a list of the tracked bundles
+	// (bundles specifically installed by the user)
+	tracked_bundles = bundle_list_tracked();
+	if (!tracked_bundles) {
+		// this should never happen, swupd_init makes sure of it
+		UNEXPECTED();
```

Comment:
The idea when I created the UNEXPECTED macro was for some unexpected behaviors that we try to recover and that wouldn't be able to be noticed unless we have an abort() call to warn us. If it's a situation where you return an error code, you don't need to use that. Just return the error code and the test is going to break.

Reply:
got it